helix-term/src/ui/mod.rs

```diff
@@ -85,10 +85,15 @@ pub fn file_picker(root: PathBuf) -> Picker<PathBuf> {
         Err(_err) => None,
     });
 
-    const MAX: usize = 8192;
+    let files = if root.join(".git").is_dir() {
+        files.collect()
```

Comment:
I wonder if we should check the parent directory as well in case they are not doing it in parent directory.